lib/_http_incoming.js

```diff
@@ -315,6 +315,8 @@ IncomingMessage.prototype._dump = function _dump() {
   if (!this._dumped) {
     this._dumped = true;
     this.resume();
+    // Flush the existing buffered data
+    this._readableState._flush();
```

Comment:
Thanks @mscdex and @mcollina. I updated the commit with `removeAllListeners('data')`.